core/src/network/reducers/publish.rs

```diff
@@ -129,7 +152,11 @@ fn reduce_publish_inner(
     let entry_with_header = fetch_entry_with_header(&address, root_state)?;
     match entry_with_header.entry.entry_type() {
         EntryType::AgentId => publish_entry(network_state, &entry_with_header),
-        EntryType::App(_) => publish_entry(network_state, &entry_with_header).and_then(|_| {
+        EntryType::App(_) => publish_entry(network_state, &entry_with_header)
+        .and_then(|_|{
+            publish_header(network_state, &entry_with_header.header)
```

Comment:
The trouble with doing this here is this publish call is only made for public entries. Need to move it elsewhere